exotica_core/CMakeLists.txt

```diff
@@ -1,6 +1,11 @@
 cmake_minimum_required(VERSION 2.8.12)
 project(exotica_core)
 
+# TODO: Start fixing warnings: - https://stackoverflow.com/a/9862800
+# -pedantic -Wall -Wextra -Wcast-align -Wcast-qual -Wctor-dtor-privacy -Wdisabled-optimization -Wformat=2 -Winit-self -Wlogical-op -Wmissing-declarations -Wmissing-include-dirs -Wnoexcept -Wold-style-cast -Woverloaded-virtual -Wredundant-decls -Wshadow -Wsign-conversion -Wsign-promo -Wstrict-null-sentinel -Wstrict-overflow=5 -Wswitch-default -Wundef -Werror -Wno-unused
+
+add_compile_options(-Wdeprecated-declarations -Woverloaded-virtual)# -Werror) # -Wall -Wextra) # -pedantic) # -Werror) # can't use -Werror due to MoveIt / ROS-Console
```

Comment:
Do we really want to turn deprecation warnings into errors? It basically has the same effect as not to use deprecation attribute, i.e. the compilation will fail because of mismatching API.

Reply:
Not sure about deprecation - but for others over time, yes. Also, I haven't yet added the `-Werror` flag.